server/sandbox_run.go

```diff
@@ -151,6 +145,28 @@ func (s *Server) RunPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
 		return nil, err
 	}
 
+	// add log directory
+	logDir := req.GetConfig().GetLogDirectory()
+	if logDir == "" {
+		logDir = filepath.Join(s.config.LogDir, id)
+	}
+	if err = os.MkdirAll(logDir, 0700); err != nil {
+		return nil, err
+	}
+	if !filepath.IsAbs(logDir) {
+		var wd string
+		logrus.Warnf("requested logDir for sbox id %s is a relative path: %s", id, logDir)
+
+		// XXX: What is a sane thing to do here if kubernetes really intended
```

Comment:
If it's something that "should never happen", I'm +1 to @runcom's comment: we should error out